core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -364,10 +364,7 @@ class KRaftMetadataCache(val brokerId: Int) extends MetadataCache with Logging w
     _currentImage.topics().topicsByName().containsKey(topicName)
 
   override def contains(tp: TopicPartition): Boolean = {
-    Option(_currentImage.topics().getTopic(tp.topic())) match {
-      case None => false
-      case Some(topic) => topic.partitions().containsKey(tp.partition())
-    }
+    Option(_currentImage.topics().getTopic(tp.topic())).exists(topic => topic.partitions().containsKey(tp.partition()))
```

Comment:
FYI reviewers:
Usage of Option.exists. No extra allocations as mentioned in an earlier comment